### DISCUSSION TITLE
Energy Module on Berry don't handle the new 8 phase energy driver on ESP32

### DISCUSSION THREAD

**Discussion by ZioFabry** [arendst/Tasmota#21871] [Support request] [answered]
Hello, the actual berry implementation of the energy module don't handle the new 8-phase energy driver, only the 3 phase "values" are returned.

I can propose a PR for that... but I'm not sure what is the best way to do.

Any suggestions ?

**Comment by s-hadinger** [comment]
I'm not completely satisfied with my current mapping to the Energy driver. What's your plan for a PR ?

**Reply by ZioFabry** [reply]
the much simple way is just to add #ifdef ESP32 at the end of the current array to add the missing 5 phases attributes... but honestly I don't like too much

**Reply by s-hadinger** [reply]
Berry is only on ESP32, so no need for `#ifdef ESP32` anyways

**Reply by ZioFabry** [reply]
Right.... so just to add 100 more attributes.... from 88 to 188... what do you think about ?

**Reply by ZioFabry** [reply]
something like that.... https://github.com/Biomine-Sustainable-Mining/Tasmota/commit/00bf91c847381e6d5fc62111b92969af6d84e9f5

**Reply by s-hadinger** [reply]
Honestly I'm not happy with that. I would rather have the module returning sub-modules or instance, that would act like arrays so we can have any number of phases, and also make it easier to iterate from code.

**Reply by s-hadinger** [reply]
Let me try something

**Reply by ZioFabry** [reply]
OK.... let me know if I can help in some way.... in the while I tested my patch and is working fine... but I agree with you... it's not elegant to hard code is this way...

**Reply by s-hadinger** [reply]
I have a working prototype, needs some more polishing

**Reply by ZioFabry** [reply]
great to know.... keep me in touch

**Comment by s-hadinger** [comment, marked as answer]
Please try latest version in development: https://github.com/arendst/Tasmota/pull/21887

**Reply by ZioFabry** [reply]
chapeau... it's work perfectly!

Thanks a lot

**Reply by ZioFabry** [reply]
My Berry dashboard ... 
![image](https://github.com/user-attachments/assets/1db66202-9bdd-452b-9fbd-df9ebeafe412)

**Reply by s-hadinger** [reply]
Nice
